modules/highgui/src/window_framebuffer.cpp

```diff
@@ -0,0 +1,767 @@
+#include "window_framebuffer.hpp"
+
+#include <opencv2/core/utils/configuration.private.hpp>
+#include <opencv2/core/utils/logger.defines.hpp>
+#ifdef NDEBUG
+#define CV_LOG_STRIP_LEVEL CV_LOG_LEVEL_DEBUG + 1
+#else
+#define CV_LOG_STRIP_LEVEL CV_LOG_LEVEL_VERBOSE + 1
+#endif
+#include <opencv2/core/utils/logger.hpp>
+
+#include <unistd.h>
+#include <stdio.h>
+#include <termios.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <linux/fb.h>
+#include <linux/input.h>
+#include <sys/mman.h>
+#include <sys/ioctl.h>
+
+#include "opencv2/imgproc.hpp"
+
+#include "XWDFile.h"
+
+namespace cv { namespace highgui_backend {
+
+    std::shared_ptr<UIBackend> createUIBackendFramebuffer()
+    {
+        return std::make_shared<FramebufferBackend>();
+    }
+
+    static std::string& getFBMode()
+    {
+        static std::string fbModeOpenCV =
+        cv::utils::getConfigurationParameterString("OPENCV_HIGHGUI_FB_MODE", "");
+        static std::string fbModeDef = "FB";
+
+        if (!fbModeOpenCV.empty()) return fbModeOpenCV;
+        return fbModeDef;
+    }
+
+    static std::string& getFBFileName()
+    {
+        static std::string fbFileNameFB =
+        cv::utils::getConfigurationParameterString("FRAMEBUFFER", "");
+        static std::string fbFileNameOpenCV =
+        cv::utils::getConfigurationParameterString("OPENCV_HIGHGUI_FB_DEVICE", "");
+        static std::string fbFileNameDef = "/dev/fb0";
+
+        if (!fbFileNameOpenCV.empty()) return fbFileNameOpenCV;
+        if (!fbFileNameFB.empty()) return fbFileNameFB;
+        return fbFileNameDef;
+    }
+
+
+    FramebufferWindow::FramebufferWindow(FramebufferBackend &_backend, int _flags):
+        backend(_backend), flags(_flags)
+    {
+        CV_LOG_INFO(NULL, "UI: FramebufferWindow::FramebufferWindow()");
+        FB_ID = "FramebufferWindow";
+        windowRect = Rect(0,0, backend.getFBWidth(), backend.getFBHeight());
+    }
+
+    FramebufferWindow::~FramebufferWindow()
+    {
+        CV_LOG_INFO(NULL, "UI: FramebufferWindow::~FramebufferWindow()");
+    }
+
+    void FramebufferWindow::imshow(InputArray image)
+    {
+        currentImg = image.getMat().clone();
```

Comment:
data race is possible here.

Why do we need such global variable?

Reply:
currentImg local variable for Windows.
currentImg is used to display the image when Windows is resized and moved.